tensorboard/webapp/util/types.ts

```diff
@@ -19,3 +19,9 @@ limitations under the License.
 export type DeepReadonly<T> = {
   readonly [P in keyof T]: DeepReadonly<T[P]>;
 };
+
+export type DeepPartial<T> = T extends object
```

Comment:
useful

Reply:
I'm surprised to discover that you can branch type definitions based on conditions on the type parameters . Kind of neat.